solutions/usaco-416.mdx

```diff
@@ -0,0 +1,86 @@
+---
+id: usaco-416
+source: USACO Silver 2014 March
+title: The Lazy Cow
+author: Brad Ma
+---
+
+[Official Analysis (C++)](http://www.usaco.org/current/data/sol_lazy_silver.html)
+
+## Implementation
+
+**Time Complexity** $\mathcal{O}(K \cdot N ^ 2)$
+
+<LanguageSection>
+  
+<JavaSection>
+  
+<Warning>
+Depending on how you implement this problem, Java code may have trouble 
+passing the last testcase even though the algorithm is correct.
+</Warning>
+  
+```java
+import java.io.*;
+import java.util.StringTokenizer;
+
+public class TheLazyCow {
+	public static void main (String[] args) throws IOException {
+		Kattio io = new Kattio("lazy");
+		int n = io.nextInt();
+		int k = io.nextInt();
+		int[][] matrix = new int[2 * n + 1][2 * n + 1];
```

Comment:
better variable names

Reply:
aren't n and k from the problem?

Reply:
it would be nice if people didn't have to constantly look back & forth at the problem & solution figuring out which variable means which thing

Reply:
usually for input values like these i think one letter names are fine and most of the solutions do it like this

Reply:
just because everyone else does it doesn't mean it's good

Reply:
i think the fact that everyone else does it means that it's up to the solution's author to decide what to do
so if you want to name n and k something different in your solutions, you can, but other people can choose to name them n and k

Reply:
honestly as a general rule of thumb, we shouldn't be too strict on one-letter variable names as long as it represents what the problem states. for @SansPapyrus683's argument regarding looking "back & forth" --> this isn't necessarily the case for most people lol. if you read the problem statement carefully and understand what it's saying, these one-letter var names should be relatively straightforward...

Reply:
who's the "most people"
i have to do every time lmao

Reply:
> who's the "most people"

people who read the problem carefully :rofl: and have basic memory

Reply:
also, $n$ is common for the length of the array / string, $m$ is commonly used when there is a 2d grid of some sort, and $a$ is commonly used as the name of the array.

Reply:
wow
all i do is just look at what they mean and go